app/controllers/moments_controller.rb

```diff
@@ -109,9 +109,9 @@ def moment_params
   end
 
   def set_association_variables!
-    @categories = current_user.categories.order(created_at: :desc)
+    @categories = current_user.categories.is_visible.or(Category.where(id: @moment.category_ids)).order(created_at: :desc)
```

Comment:
I found a bug:

1. Create a Moment with a visible Category
2. Edit the Category to be not visible
3. Edit the Moment and uncheck the Category and hit "Submit"
4. The Category is not removed (it should be removed)

Reply:
The issue is that if we're editing a Moment or Strategy, we're overriding tags that are removed before they are saved.